Model/Checkout/ShippingMethods.php

```diff
@@ -11,32 +12,70 @@
 class ShippingMethods implements ShippingMethodsInterface
 {
     /**
-     * @param mixed $deliveryOptions
+     * @var \Magento\Checkout\Model\Session
+     */
+    private $session;
+
+    /**
+     * ShippingMethods constructor.
+     *
+     * @param \Magento\Checkout\Model\Session $session
+     */
+    public function __construct(Session $session)
+    {
+        $this->session = $session;
+    }
+
+    /**
+     * @param array[] $deliveryOptions indexed array holding 1 deliveryOptions object
      *
-     * @return mixed[]
+     * @return array[]
      * @throws Exception
      */
-    public function getFromDeliveryOptions($deliveryOptions)
+    public function getFromDeliveryOptions(array $deliveryOptions): array
     {
-        if (! $deliveryOptions[0]) {
+        if (! $deliveryOptions) {
             return [];
         }
 
+        $deliveryOptions = $deliveryOptions[0];
```

Comment:
Zou je de `0` kunnen vervangen door een constante, zodat het gelijk duidelijk is dat het om de "echte" delivery options gaat?

Reply:
Ik denk niet dat dat iets toevoegt hier, de beschrijving bij de @param vind ik wel goed zo